### PR TITLE
fix(mount_matrix): Update gyro matrices

### PR DESCRIPTION
**Pull request by uejji**
~~WIP.  Still gathering device data.~~

### CODE REVIEW

**Comment by uejji** [comment]
Per discussion with @pastaq, this is a step towards bringing InputPlumber's gyro output in line with that of Steam Deck.

**Comment by pastaq** [comment]
I'll test this on some of the remaining devices this weekend.

**Comment by uejji** [comment]
Please feel free to doublecheck my submissions, too.

For reference I've attached an image to illustrate which rotations produce positive and negative deflection values on Steam Deck.  Follow the marked arrow (eg clockwise yaw produces positive values).

Untitled

File: rootfs/usr/share/inputplumber/devices/50-ayaneo_2021.yaml [hunk 1]
```diff
@@ -48,9 +48,10 @@ source_devices:
     iio:
       name: i2c-10EC5280:00
       mount_matrix:
+        # TODO: Reverify on actual hardware
         x: [0, -1, 0]
-        y: [0, 0, -1]
-        z: [1, 0, 0]
+        y: [1, 0, 0]
```
Comment (pastaq) [inline rootfs/usr/share/inputplumber/devices/50-ayaneo_2021.yaml:53]: Based on your diagram of positive/negative deflection, this should be inverted.
```suggestion
        y: [-1, 0, 0]
```

File: rootfs/usr/share/inputplumber/devices/50-ayn_loki_max.yaml [hunk 2]
```diff
@@ -36,9 +36,11 @@ source_devices:
     iio:
       name: i2c-BMI0160:00
       mount_matrix:
-        x: [1, 0, 0]
-        y: [0, 0, -1]
-        z: [0, 1, 0]
+        # Calculated from observed gyro output
+        # TODO: Verify updated matrix on actual hardware
+        x: [-1, 0, 0]
```
Comment (pastaq) [inline rootfs/usr/share/inputplumber/devices/50-ayn_loki_max.yaml:41]: Based on your diagram of positive/negative deflection, these should be inverted.
```suggestion
        x: [1, 0, 0]
```

File: rootfs/usr/share/inputplumber/devices/50-ayn_loki_max.yaml [hunk 3]
```diff
@@ -36,9 +36,11 @@ source_devices:
...
+        y: [0, -1, 0]
+        z: [0, 0, 1]
```
Comment (pastaq) [inline rootfs/usr/share/inputplumber/devices/50-ayn_loki_max.yaml:43]:
```suggestion
        z: [0, 0, -1]
```

**Review by pastaq — CHANGES_REQUESTED** [review]
I was able to test the following devices:
AYANEO 2021
AYANEO 2
AYANEO AIR
AYANEO AIR Plus
AYN Loki Max

I tried to test the AYANEO Next Lite but it was behaving very strangely and I wasn't able to get a valid test out of it. Similarly I need to reinstall the OS on my OrangePi Neo as its currently stuck on a broken version. I'll be unavailable for about 2 weeks if anyone else wants to provide feedback for the remaining devices.

I'd like to double check again that the axes in your last comment are correct for positive/negative values. The animations feel backwards for the first and second axes when the correct matrix is applies to render positive/negative values as described in it.

**Comment by uejji** [comment]
This isn't the best example, but I think it can help if reviewed at 0.5x speed:
https://youtu.be/1TL8WtX_IXs?t=80

This video at the marked timestamp shows which rotations on the Steam Deck produce positive or negative deflection values for the first two axes.

I struggle to find an accurate and well-communicated method of describing these rotations other than the image provided in the previous comment, but I would definitely welcome any other descriptions.

If your corrections correlate to the rotations and deflection values shown in the video, then they are correct.

**Comment by pastaq** [comment]
Okay, seems we're on the same page then.

I'll finish testing the remaining devices later this month. Thanks.

**Comment by uejji** [comment]
Thanks to:
@pastaq for providing gyro testing on Air, 2021 and Loki Max
@winghugs for providing gyro testing on Air 1S 8840U

**Comment by pastaq** [comment]
I should have some time this weekend to verify on the following devices:
AYANEO Flip KB
Legion Go
AYANEO 2S
OrangePi Neo
AYANEO Next Lite

That should cover the majority of devices people are using. It would be nice to get an Loki Zero and Loki Mini Pro, air plus mendo before this is merged. I'm comfortable with the win4, air 1s as they are tested on at leas t one CPU. Claw I'm not worries about as that hasn't been confirmed working yet anyway.

**Comment by uejji** [comment]
> Loki Zero and Loki Mini Pro, air plus mendo

No idea if I'll be able to get access to any of those, but I'll update if someone with one turns up.

**Comment by fewtarius** [comment]
> I should have some time this weekend to verify on the following devices: AYANEO Flip KB Legion Go AYANEO 2S OrangePi Neo AYANEO Next Lite
> 
> That should cover the majority of devices people are using. It would be nice to get an Loki Zero and Loki Mini Pro, air plus mendo before this is merged. I'm comfortable with the win4, air 1s as they are tested on at leas t one CPU. Claw I'm not worries about as that hasn't been confirmed working yet anyway.

We spent some time today on the Flip KB and 2S to make sure they behave correctly. :)

**Comment by uejji** [comment]
Tested successfully on Flip DS

**Review by pastaq — APPROVED** [review]
I think this is well baked. Any issues can be solved in a bug fix.

**Comment by uejji** [comment]
Resolves https://github.com/ShadowBlip/InputPlumber/issues/217

**Pull request merged**